base/set.jl

```diff
@@ -191,6 +191,37 @@ function unique(f::Callable, C)
 end
 
 """
+	unique!(itr)
+
+Remove duplicate items and return the modified collection,
+as determined by [`isequal`](@ref).
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia>  unique!([7, 3, 2, 3, 7, 5])
```

Comment:
Right, I thought that we were using the version with less allocation, that was what my pull request was for. This is my first time contributing so I think that I may have done some things wrong with my pull request and stepped slightly outside my skill range.

Would you say that the right thing to do is to just make a new pull request with the Jack/Pablo version of `unique!` with the updated documentation and adding `unique!` to base/exports.jl?